### DISCUSSION TITLE
Blocked Internet C200 keeps resetting itself

### DISCUSSION THREAD

**Discussion by brianjmurrell** [JurajNyiri/HomeAssistant-Tapo-Control#662] [General]
After doing a firmware update earlier today I am noticing that my Internet-blocked[1] C200 is resetting (where it clicks and then rotates, moves the lens up and down, etc.) every few minutes.  The video stream also tends to freeze frequently.

Tapo app says my firmware is `Version:1.3.14 Build 240513 Rel 43636n(5553)`.

[1] I block all outgoing access to the Internet at the router to prevent any possibility of somebody hacking TP-Link's Cloud control-centre and getting into my camera and it's video feed.

**Comment by JurajNyiri** [comment]
Unfortunately there is nothing that can be done here. If the camera resets without integration being used it’s a camera issue and you need to open ticket with tplink to fix the issue via firmware.

**Reply by brianjmurrell** [reply]
Understood.  I think this was more of a shout-out/sanity check to see if anyone else had any similar experiences.

**Comment by wilcomir** [comment]
Hello, I have a similar problem with my camera, and I do not even use this integration - I use the camera through frigate.

This is both to confirm to OP that he is not alone, and that the issue here does not lie within the integration, but most probably somewhere else.

**Comment by manunited10** [comment]
Does it happen when trying to move the camera (PTZ) or just randomly reboots?

**Comment by wilcomir** [comment]
It is quite hard to pinpoint this issue @manunited10 - to me it seems that it happens when I try to PTZ, but not always. I have an automation in home assistant that triggers both on some event and on a time pattern; I had to add the time pattern because commanding a PTZ sometimes fails.
It seems to me that when it fails, it times out - hard to debug as the stack trace I see in HA is quite cryptic - and then it resets. I have no idea why this happens, the only other thing I might want to add is that my camera is firewall blocked, and apparently its datetime drifts significantly - I am not sure whether this has an impact or not.

By the way keep in mind that I do not use this integration, I have my camera connected to a frigate instance, and then exposed to home assistant. I believe what happens is that sometimes the PTZ command is either malformed, or bad-received, and this triggers a timeout on the calling function on HA side, and a reset on the camera side.

Where the issue occurs is very difficult to say though.

**Reply by manunited10** [reply]
I'm experiencing the same-ish issue with three C210 cameras that I own. The cameras are blocked from the internet, and the video stream works fine. However, when I use PTZ (either through this integration, my Blue Iris NVR, or third-party software like iSpy), the cameras sometimes move briefly before crashing and rebooting.

I've been in contact with TP-Link and have sent them some logs. They recently created a test firmware for one of my cameras (linked to its MAC address), and my initial tests suggest that it has resolved the issue. They mentioned that they will provide this firmware for my other cameras later. I guess it may eventually be released publicly.

**Reply by wilcomir** [reply]
This is very interesting.

In case you want to mention it to them, I also have a C210 and am happy to validate the firmware as well 😁

**Reply by manunited10** [reply]
It's an ota update and I had to connect my camera to the internet to receive it.

**Reply by wilcomir** [reply]
What version is it?

**Reply by manunited10** [reply]
Version:1.4.2 Build 240910 Rel.38804n

**Reply by wilcomir** [reply]
I am on 1.3.14 Build 240801 so I guess I will have to wait. Thanks for the info though, really helpful!

**Reply by manunited10** [reply]
They mentioned today that they are planing to release 1.4.2 this month. So hopefully you get it in a few days. It all started [here](https://community.tp-link.com/en/smart-home/forum/topic/666194) btw.
